### PR TITLE
Implement Into<Vec<Message>> for QueryError and QueryEvent (#160)

### LINKED ISSUE #160: Implement Into<Vec<Message>> for QueryError and QueryEvent
Currently, `QueryError` has `into_message` function and `QueryEvent` is mapped to `Vec<Message>` in `QueryResultMapper`. The good way to unify is to implement `Into<Vec<Message>>` trait

### PR DESCRIPTION
**Pull request by Tolledo**
Closes #160 

#### Description
Implement Into<Vec<Message>> for QueryError and QueryEvent. Got rid of QueryResultMapper

### CODE REVIEW

File: src/protocol/src/lib.rs [hunk 1]
```diff
@@ -130,10 +130,11 @@ impl<RW: AsyncReadExt + AsyncWriteExt + Unpin> Connection<RW> {
         }
     }
 
-    /// Sends response messages to client. Most of the time it is a single
+    /// Sends response messages to client. Most of the time it is a single 
     /// message, select result one of the exceptional situation
     pub async fn send(&mut self, query_result: QueryResult) -> io::Result<()> {
-        for message in QueryResultMapper::map(query_result) {
+        let messages: Vec<Message> = query_result.map_or_else(|event| event.into(), |err| err.into());
```
Comment (Tolledo) [inline src/protocol/src/lib.rs:136]: Needs advice: is it the best way of mapping both Ok() and Err() to the same type?

Comment (AndrewBregger) [reply src/protocol/src/lib.rs:136]: I would have implement Into<Vec<Message>> for QueryResult. That would make the above line `query_result.into()`.  The implementation would have been a simple pattern match.

I'm curious what others think.

Comment (alex-dukhno) [reply src/protocol/src/lib.rs:136]: `QueryResult` is type alias `pub type QueryResult = std::result::Result<QueryEvent, QueryError>;`
Rust does not allow to implement `trait` that out of your crate, `Into` is in `std`, for `struct`, `enum` that is out of your crate, `Result` is in `std`.

So the best way to do is either
```rust
match query_result {
  Ok(event) => event.into(),
  Err(error) => error.into()
}
```
or as you did.
You can try with match and see if clippy will complain ... I'd go with `match` I think, it is more explicit

Comment (Tolledo) [reply src/protocol/src/lib.rs:136]: Agreed, `map_or_else` is kinda cool, but not raising a bar in term of readability

**Pull request merged**